### PR TITLE
Set publisherId=loggedUserId of new Places/area stream and relate to global Places/location with publisherId=UBI.

### PR DESCRIPTION
**Pull request by AndreyTepaykin**

### CODE REVIEW

File: platform/plugins/Places/web/js/tools/location.js [hunk 1]
```diff
@@ -190,11 +198,10 @@ Q.Tool.define("Places/location", function (options) {
...
 								longitude: result.geometry.location.lng(),
 								locationType: result.geometry.type,
-								venue: place.name
+								venue: place.name,
+								placeId: result.place_id
 							};
-							if (result.place_id) {
-								attributes.placeId = result.place_id;
-							}
+
```
Comment (EGreg) [inline platform/plugins/Places/web/js/tools/location.js:206]: Why did you remove this code and just put it up there without the if? 

 							if (result.place_id) {	 
  								attributes.placeId = result.place_id;	
 							}

File: platform/plugins/Places/web/js/tools/location.js [hunk 2]
```diff
@@ -205,31 +212,38 @@ Q.Tool.define("Places/location", function (options) {
 									if (!title) {
 										return;
 									}
-									var publisherId = state.publisherId || userId;
```
Comment (EGreg) [inline platform/plugins/Places/web/js/tools/location.js:208]: Where did this code go?

File: platform/plugins/Places/web/js/tools/location.js [hunk 3]
```diff
@@ -205,31 +212,38 @@ Q.Tool.define("Places/location", function (options) {
...
-											var sf = this.fields;
-											tool.relatedTool.state.onRefresh.setOnce(
-											function (previews, map, entering, exiting, updating) {
-												var key = Q.firstKey(entering);
-												var index = map[key];
-												var preview = previews[index];
-												Q.Pointer.canceledClick = false;
-												tool.toggle(preview.element);
-											});
+
```
Comment (EGreg) [inline platform/plugins/Places/web/js/tools/location.js:217]: What is happening here?

File: platform/plugins/Places/web/js/tools/location.js (hunk 1 above)
Comment (AndreyTepaykin) [reply platform/plugins/Places/web/js/tools/location.js:206]: 1) in Places/location/post.php I use Places_Location::stream to get or create location stream, and this method require placeId.
2) as I see, any google location return placeId, even just by lat,lng

File: platform/plugins/Places/web/js/tools/location.js (hunk 3 above)
Comment (AndreyTepaykin) [reply platform/plugins/Places/web/js/tools/location.js:217]: I tried to set Places/location/ template for main community with all permissions=40 but you told that this  unacceptable. 
So I modified Places/location streams creation.
Instead of creation stream on client, send request to backend Places/location POST handler and create with option 'skipAccess'=true.
This need to avoid exception NotAuthorized when user trying to create location stream on behalf of main community.

File: platform/plugins/Places/web/js/tools/location.js (hunk 1 above)
Comment (EGreg) [reply platform/plugins/Places/web/js/tools/location.js:206]: Let's put the `if` just in case, [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming)

File: platform/plugins/Places/web/js/tools/location.js (hunk 3 above)
Comment (EGreg) [reply platform/plugins/Places/web/js/tools/location.js:217]: Why? The access control of Places/location template is exactly to express who can do what. Anyone being able to do anything ... this is permissions 40 for everyone, right? I like that you found a way to do it in Qbix.  But instead, let's go back to client code like before, and simply express access the following way:

https://qbix.com/platform/guide/access#creating

Read it carefully, about Template A and Template B and relation between them, this way we can allow any user to create location stream related to main Communities locations index by certain relation type, probably called "Places/locations".

Notice that in Template B, the user does not need to have 40 40 40 permissions. They can have *any* limited permissions. But they are still allowed to create a related document.

Why do we do this? Well let's say we have a shared folder, and people are given access to create files in it. They can create a file, and we can precisely set their access to the file after creating.

I think the public access should be readLevel 40 (anyone can see location created in publc index), writeLevel 0 (public cannot edit location stream) and adminLevel 20 (can invite to location stream). "Places/admins" role of main community should be able to have 40/40/40 so they can even delete locations.

Later, people will be able to create locations in communities, which may not be public. (Like a private building community, that not everyone has access to.) But for now it's all public.

Similarly, for areas. Have a Template A for locations and Template B for areas, and Qbix will allow client to create areas related to locations. With same public access as above.

Comment (AndreyTepaykin) [reply platform/plugins/Places/web/js/tools/location.js:217]: Yes you right. Seems I confused, sorry.
Return creation of location stream back to client (look last commits).
Tested, works fine.

**Review by EGreg — APPROVED** [review]
